lib/Doctrine/DBAL/Driver/OCI8/OCI8Statement.php

```diff
@@ -301,18 +303,35 @@ public function bindParam($column, &$variable, $type = ParameterType::STRING, $l
             $lob = oci_new_descriptor($this->_dbh, OCI_D_LOB);
             $lob->writeTemporary($variable, OCI_TEMP_BLOB);
 
-            $this->boundValues[$column] =& $lob;
-
-            return oci_bind_by_name($this->_sth, $column, $lob, -1, OCI_B_BLOB);
-        } elseif ($length !== null) {
-            $this->boundValues[$column] =& $variable;
-
-            return oci_bind_by_name($this->_sth, $column, $variable, $length);
+            $variable =& $lob;
         }
 
         $this->boundValues[$column] =& $variable;
 
-        return oci_bind_by_name($this->_sth, $column, $variable);
+        return oci_bind_by_name(
+            $this->_sth,
+            $column,
+            $variable,
+            $length ?? -1,
+            $this->convertParamType($type)
+        );
+    }
+
+    /**
+     * Converts DBAL parameter type to oci8 parameter type
+     */
+    private function convertParamType(int $type) : int
+    {
+        switch ($type) {
```

Comment:
This could also be written as:
```php
return [
    ParameterType::BINARY => OCI_B_BIN,
    ParameterType::LARGE_OBJECT => OCI_B_BLOB,
][$type] ?? SQLT_CHR;
```
Up to you which variant you prefer.

Reply:
I'll keep it implemented using `switch`. In PHP 7, the lookup is the same efficient as using an array, it's more maintainable (you can easily add logic per key) and it doesn't initialize a temporary array upon each call.